<ui version="4.0" >
 <class>SnipetDialog</class>
 <widget class="QDialog" name="SnipetDialog" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>596</width>
    <height>595</height>
   </rect>
  </property>
  <property name="windowTitle" >
   <string>Template property</string>
  </property>
  <layout class="QGridLayout" >
   <item row="0" column="0" >
    <widget class="QGroupBox" name="groupBox" >
     <property name="title" >
      <string>&amp;Description</string>
     </property>
     <layout class="QGridLayout" >
      <property name="margin" >
       <number>9</number>
      </property>
      <property name="spacing" >
       <number>6</number>
      </property>
      <item row="4" column="0" >
       <widget class="QLabel" name="label_4" >
        <property name="text" >
         <string>&amp;Category :</string>
        </property>
        <property name="buddy" >
         <cstring>m_categoryComboBox</cstring>
        </property>
       </widget>
      </item>
      <item row="1" column="0" >
       <widget class="QLabel" name="label_2" >
        <property name="text" >
         <string>&amp;Icon :</string>
        </property>
        <property name="buddy" >
         <cstring>m_iconLineEdit</cstring>
        </property>
       </widget>
      </item>
      <item row="0" column="0" >
       <widget class="QLabel" name="label" >
        <property name="text" >
         <string>&amp;Name :</string>
        </property>
        <property name="buddy" >
         <cstring>m_nameLineEdit</cstring>
        </property>
       </widget>
      </item>
      <item row="5" column="2" >
       <widget class="QPlainTextEdit" name="m_descriptionTextEdit" >
        <property name="sizePolicy" >
         <sizepolicy vsizetype="Maximum" hsizetype="Expanding" >
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximumSize" >
         <size>
          <width>16777215</width>
          <height>100</height>
         </size>
        </property>
        <property name="whatsThis" >
         <string>Long descriptions of the snipet.</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2" >
       <widget class="QLineEdit" name="m_iconLineEdit" >
        <property name="whatsThis" >
         <string>Path to an icon for the template. If the path begin by :/ the icon is looked in the &lt;b>XINX&lt;/b> resources</string>
        </property>
       </widget>
      </item>
      <item row="4" column="2" >
       <widget class="QComboBox" name="m_categoryComboBox" >
        <property name="whatsThis" >
         <string>Category where the snipet must be classed.</string>
        </property>
        <property name="editable" >
         <bool>true</bool>
        </property>
        <item>
         <property name="text" >
          <string>Others</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="5" column="0" colspan="2" >
       <widget class="QLabel" name="label_3" >
        <property name="text" >
         <string>&amp;Description :</string>
        </property>
        <property name="alignment" >
         <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
        </property>
        <property name="buddy" >
         <cstring>m_descriptionTextEdit</cstring>
        </property>
       </widget>
      </item>
      <item row="0" column="2" >
       <widget class="QLineEdit" name="m_nameLineEdit" >
        <property name="whatsThis" >
         <string>Name of the snipet used in to identify the snipet in menus.</string>
        </property>
       </widget>
      </item>
      <item row="3" column="2" >
       <widget class="QLineEdit" name="m_keyLineEdit" >
        <property name="whatsThis" >
         <string>Word key used to call the snipet in the &lt;b>XINX&lt;/b> editor. If no key is defined, it's not possible to call the template in the completion.</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0" >
       <widget class="QLabel" name="label_5" >
        <property name="text" >
         <string>&amp;Snipet key :</string>
        </property>
        <property name="buddy" >
         <cstring>m_keyLineEdit</cstring>
        </property>
       </widget>
      </item>
      <item row="2" column="0" >
       <widget class="QLabel" name="label_6" >
        <property name="text" >
         <string>&amp;Extention :</string>
        </property>
        <property name="buddy" >
         <cstring>m_extLineEdit</cstring>
        </property>
       </widget>
      </item>
      <item row="2" column="2" >
       <widget class="QLineEdit" name="m_extLineEdit" >
        <property name="whatsThis" >
         <string>Extention to use for the syntax highlighter in the editor below.</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="1" >
    <widget class="QGroupBox" name="m_paramGroupBox" >
     <property name="minimumSize" >
      <size>
       <width>250</width>
       <height>0</height>
      </size>
     </property>
     <property name="whatsThis" >
      <string>List of name parameter used by the template. The first parameter replace the key word %1, the second parameter replace the key word %2, ...</string>
     </property>
     <property name="title" >
      <string>&amp;Parameter</string>
     </property>
    </widget>
   </item>
   <item row="1" column="0" colspan="2" >
    <widget class="QGroupBox" name="m_templateGroupBox" >
     <property name="title" >
      <string>&amp;Template text :</string>
     </property>
     <layout class="QGridLayout" >
      <item row="0" column="0" >
       <widget class="XinxCodeEdit" native="1" name="m_textEdit" >
        <property name="whatsThis" >
         <string>Templat to show to screen. In this template you can define parameter by used &lt;i>%n&lt;/i> keyword where &lt;i>n&lt;/i> is a number. More sophisticate snipet can use ECMAScript. Balise &lt;i>&amp;lt;?= ?&amp;gt;&lt;/i> is used to call script and show the text returned by the script. &lt;i>&amp;lt;?    ?&amp;gt;&lt;/i> can be used to init the script and make some pre-process.</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="0" colspan="2" >
    <widget class="QDialogButtonBox" name="buttonBox" >
     <property name="standardButtons" >
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>XinxCodeEdit</class>
   <extends>QWidget</extends>
   <header location="global" >xinxcodeedit.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>m_nameLineEdit</tabstop>
  <tabstop>m_iconLineEdit</tabstop>
  <tabstop>m_extLineEdit</tabstop>
  <tabstop>m_keyLineEdit</tabstop>
  <tabstop>m_categoryComboBox</tabstop>
  <tabstop>m_descriptionTextEdit</tabstop>
  <tabstop>m_textEdit</tabstop>
 </tabstops>
 <resources>
  <include location="../application.qrc" />
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>SnipetDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>459</x>
     <y>574</y>
    </hint>
    <hint type="destinationlabel" >
     <x>594</x>
     <y>272</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>SnipetDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>313</x>
     <y>569</y>
    </hint>
    <hint type="destinationlabel" >
     <x>593</x>
     <y>216</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
